.github/CONTRIBUTING.md

```diff
@@ -81,6 +99,20 @@ $ make html
 
 The built documentation can then be found in `docs/_build/html/`.
 
+To file a pull request, create a new branch on top of the upstream repository:
+
+```console
+$ git checkout -b my_topical_branch upstream/main
+```
+
+Make your changes, push them to your fork (the remote *origin*) and publish the PR!
+
+Before starting to work on your next pull request, run the following command to sync your local repository with the remotes:
```

Comment:
This is a bit inaccurate. The local repository does not absorb the tree from upstream. Fetch only updates a local copy of the remotes, separately from the local Git repository tree. We could also tell people that they can sync up their remote fork after fetching with something like `git push origin upstream/main:main` but that's probably an overkill.